### ISSUE TITLE
no-extra-parens options work inconsistently

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#12717]
**Tell us about your environment**

* **ESLint Version:** v.6.8.0
* **Node Version:** v12.13.0
* **npm Version:** v6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default(espree)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

* The `enforceForArrowConditionals: false` option doesn't allow extra parentheses more than one pair. - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogW1wiZXJyb3JcIiwgXCJhbGxcIiwgeyBcImVuZm9yY2VGb3JBcnJvd0NvbmRpdGlvbmFsc1wiOiBmYWxzZSB9XSAqL1xuXG5jb25zdCBiID0gYSA9PiAxID8gMiA6IDM7XG5jb25zdCBlID0gYyA9PiAoMSA/IDIgOiAzKTtcbmNvbnN0IGYgPSBjID0+ICgoKDEgPyAyIDogMykpKTsgLy8gVW5uZWNlc3NhcnkgcGFyZW50aGVzZXMgZXhwcmVzc2lvbi5cbmNvbnN0IGcgPSBjID0+ICgoKCgxID8gMiA6IDMpKSkpOyAvLyBVbm5lY2Vzc2FyeSBwYXJlbnRoZXNlcyBleHByZXNzaW9uLiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOnRydWV9fX0=)
```js
/* eslint no-extra-parens: ["error", "all", { "enforceForArrowConditionals": false }] */

const b = a => 1 ? 2 : 3;
const e = c => (1 ? 2 : 3);
const f = c => (((1 ? 2 : 3))); // Unnecessary parentheses expression.
const g = c => ((((1 ? 2 : 3)))); // Unnecessary parentheses expression.
```
 * On the other hand, the other options allow extra parentheses without limit. - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogW1wiZXJyb3JcIiwgXCJhbGxcIiwge1xuXHRcImNvbmRpdGlvbmFsQXNzaWduXCI6IGZhbHNlLFxuXHRcIm5lc3RlZEJpbmFyeUV4cHJlc3Npb25zXCI6IGZhbHNlLFxuICAgICAgICBcImlnbm9yZUpTWFwiOiBcImFsbFwiLFxuICAgICAgICBcImVuZm9yY2VGb3JTZXF1ZW5jZUV4cHJlc3Npb25zXCI6IGZhbHNlLFxuICAgICAgICBcImVuZm9yY2VGb3JOZXdJbk1lbWJlckV4cHJlc3Npb25zXCI6IGZhbHNlXG59XSAqL1xuXG53aGlsZSAoKCgoZm9vID0gYmFyKCkpKSkpIHt9XG5cbnggPSBhIHx8ICgoKCgoYiAmJiBjKSkpKSk7XG5cbmNvbnN0IGJheiA9ICgoKCg8ZGl2Lz4pKSkpO1xuXG4oKChhLCBiKSkpO1xuKChhLCBiKSk7XG4oYSwgYik7XG5cbnZhciBmb28gPSAoKChuZXcgQmFyKCkpKSkuYmF6O1xudmFyIGZvbyA9ICgobmV3IEJhcigpKSkuYmF6O1xudmFyIGZvbyA9IChuZXcgQmFyKCkpLmJhejsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJicm93c2VyIjp0cnVlfX19)
```js
/* eslint no-extra-parens: ["error", "all", {
	"conditionalAssign": false,
	"nestedBinaryExpressions": false,
        "ignoreJSX": "all",
        "enforceForSequenceExpressions": false,
        "enforceForNewInMemberExpressions": false
}] */

while ((((foo = bar())))) {}

x = a || (((((b && c)))));

const baz = ((((<div/>))));

(((a, b)));
((a, b));
(a, b);

var foo = (((new Bar()))).baz;
var foo = ((new Bar())).baz;
var foo = (new Bar()).baz;
```

**What did you expect to happen?**

I think the options should work consistently.
By the docs, the option `enforceForArrowConditionals: false` should allow all extra parentheses without limitation in matched cases with the option.
>...
"enforceForArrowConditionals": false allows extra parentheses around ternary expressions which are the body of an arrow function
...

**What actually happened? Please include the actual, raw output from ESLint.**

  

**Are you willing to submit a pull request to fix this bug?**

Yes. If this issue is a bug.  After #12697 merged.

**Comment by kaicataldo** [comment]
This seems like a bug to me. Thanks for reporting this :+1:
